src/main/scala/piconot/internal/Language.scala

```diff
@@ -0,0 +1,269 @@
+          System.out.println("Label States in first, if" + labelMap)
+          //break
+        }
+//      }
+      label.rules.foreach((rule: Rules) => {
+        System.out.println("Label States in second" + labelMap)
+        rule match {
+          case Face(dir) => counter = counter + 1
+          case _         => 0
+        }
+      })
+      
+      System.out.println(label)
+      System.out.println(label.name)
+      System.out.println(labelMap)
+      System.out.println(counter)
+      labelMap + (label.name -> counter)
+
+    })
+  }
+
+  def buildrules(rules: List[Label]) = {
+    var builtRules: List[picolib.semantics.Rule] = List()
+
+    rules.foreach((label: Label) => {
+
+      label.rules.foreach((rule: Rules) => {
+        rule match {
+          case Face(dir) => builtRules = builtRules ++ List(evalFace(dir))
+          case Go(sur)   => builtRules = builtRules ++ List(evalGo(sur))
+          case Jump(lbl) => builtRules = builtRules ++ List(evalJump(lbl))
+        }
+
+      })
+    })
+    builtRules
+  }
+
+  def evalFace(dir: Direction): Rule = {
+    direction = dir
+    stCounter += 1
+    Rule(State(stCounter - 1 + ""), Surroundings(Anything, Anything, Anything, Anything), StayHere, State(stCounter + ""))
+  }
+
+  def evalGo(sur: Surrounding): Rule = {
+    var surRule = Surroundings(Anything, Anything, Anything, Anything)
+    var dirRule: MoveDirection = North
+    if (direction == NORTH) {
+      dirRule = North
+      if (sur == RIGHT_WALL) {
+        surRule = Surroundings(Open, Open, Anything, Anything)
+      } else if (sur == LEFT_WALL) {
+        surRule = Surroundings(Open, Anything, Open, Anything)
+      } else if (sur == FRONT_WALL) {
+        surRule = Surroundings(Open, Anything, Anything, Anything)
+      } else if (sur == RIGHT_LEFT_WALL) {
+        surRule = Surroundings(Open, Open, Open, Anything)
+      } else if (sur == RIGHT_FRONT_WALL) {
+        surRule = Surroundings(Open, Open, Anything, Anything)
+      } else if (sur == RIGHT_LEFT_FRONT_WALL) {
+        surRule = Surroundings(Open, Open, Open, Anything)
+      } else if (sur == LEFT_FRONT_WALL) {
+        surRule = Surroundings(Open, Anything, Open, Anything)
+      } else if (sur == RIGHT_OPENING) {
+        surRule = Surroundings(Open, Blocked, Anything, Anything)
+      } else if (sur == LEFT_OPENING) {
+        surRule = Surroundings(Open, Anything, Blocked, Anything)
+      } else if (sur == RIGHT_LEFT_OPENING) {
+        surRule = Surroundings(Open, Blocked, Blocked, Anything)
+      } else if (sur == RIGHT_BACK_OPENING) {
+        surRule = Surroundings(Open, Blocked, Anything, Blocked)
+      } else if (sur == RIGHT_LEFT_BACK_OPENING) {
+        surRule = Surroundings(Open, Blocked, Blocked, Blocked)
+      } else if (sur == LEFT_BACK_OPENING) {
+        surRule = Surroundings(Open, Anything, Blocked, Blocked)
+      } else if (sur == BACK_OPENING) {
+        surRule = Surroundings(Open, Anything, Anything, Blocked)
+      }
```

Comment:
Hmm reading through these I really don't understand what the names are supposed to mean. For instance, I would think that LEFT_OPENING would just mean that left was open in which case, when you are going North I think that means you want to have an opening to the West.  Therefor I'd expect rules to be Anything, Anything, Open, Anything.